design/apidsl/type.go

```diff
@@ -59,6 +63,7 @@ func Type(name string, dsl func()) *design.UserTypeDefinition {
 	return t
 }
 
+// ArrayOf can be used in: (it is a top level DSL)
```

Comment:
This has the same issue as MediaType in that there is still no place to see all the places that the type identifier can be used.